rcl/test/rcl/test_expand_topic_name.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "rcl/expand_topic_name.h"
+
+#include "rcl/error_handling.h"
+
+using namespace std::string_literals;
+
+TEST(test_expand_topic_name, normal) {
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcutils_string_map_t subs = rcutils_get_zero_initialized_string_map();
+  rcutils_ret_t rcu_ret = rcutils_string_map_init(&subs, 0, allocator);
+  ASSERT_EQ(RCUTILS_RET_OK, rcu_ret);
+  ret = rcl_get_default_topic_name_substitutions(&subs);
+  ASSERT_EQ(RCL_RET_OK, ret);
+
+  // {node}/chatter example
+  {
+    const char * topic = "{node}/chatter";
+    const char * ns = "/my_ns";
```

Comment:
It is allowed when creating a node, but here (in `rcl_expand_topic_name`) I'm assuming that this is not the case so the logic does not have to be repeated:

https://github.com/ros2/rcl/blob/90176d9f1a3551deb861e698f36f81534e7763ac/rcl/src/rcl/node.c#L121